alpa/device_mesh.py

```diff
@@ -498,6 +513,93 @@ def run_allgather_task(self, uuid, buffer_uuids):
                            allgather_spec.tensor_slices,
                            allgather_spec.output_slice)
 
+    def broadcast(self, uuids, comm_key, world_size, devices_ids, devices_global_rank, tensor_slices, group_name):
+        used_tensors = []
+        slice_types = []            
+        is_bool = self.buffers[uuids[devices_ids[0]]].dtype == np.bool_
+        for device_id, global_rank, tensor_slice in zip(devices_ids, devices_global_rank, tensor_slices):
+            uuid = uuids[device_id]
+            tensor_shape = self.buffers[uuid].shape
+            slice_shape = tuple(ind.stop - ind.start for ind in tensor_slice)
+            if is_continuous_subset(tensor_slice, tensor_shape):
+                # fast path, two cases: (1) same shape, (2) continuous subset.
+                if slice_shape != tensor_shape:
+                    ind, _ = infer_offset_and_n_elements(tensor_slice)
+                    slice_types.append("continuous_subset")
+                    to_use = (xla_buffer_to_cupy(self.buffers[uuid]), ind)
+                else:
+                    slice_types.append("same_shape")
+                    to_use = xla_buffer_to_cupy(self.buffers[uuid])
+            elif global_rank == 0:
+                start_indices = tuple(o.start for o in tensor_slice)
+                to_use = jax_tensor_index(
+                    xla_buffer_to_jax_tensor(self.buffers[uuid]), start_indices,
+                    slice_shape)
+                to_use = jax_tensor_to_cupy(to_use)
+                slice_types.append("non_continuous_subset")
+            else:
+                to_use = device_put(
+                    jnp.ones(slice_shape, dtype=self.buffers[uuid].dtype),
```

Comment:
I think we've created a recv buffer by AllocZeroBufferExecutable

Reply:
Yep, here I keep this functionality that eagerly puts buffer to devices. Do you mean I could just delete it in order to keep code simple because we will never use it.

Reply:
sry, I haven't noticed the likewise one in `recv_tile`. @zhisbug I remember we do not prefer buffer allocation with jax because it always compiles and is slow. Will this line have the problem? If not, just ignore this comment.